crates/walrus-sui/src/client/retry_client/retriable_rpc_client.rs

```diff
@@ -293,58 +301,80 @@ impl RetriableRpcClient {
     ) -> Result<CheckpointData, RetriableClientError> {
         let _scope = monitored_scope("RetriableRpcClient::get_full_checkpoint");
         let start_time = Instant::now();
-        let error = match self.get_full_checkpoint_from_primary(sequence_number).await {
-            Ok(checkpoint) => {
+
+        // Check if we should directly skip RPC node due to previous failures, and use fallback.
+        let try_rpc_node_first = self.fallback_client.is_none()
+            || start_time >= self.skip_rpc_for_checkpoint_until.load(Ordering::Relaxed);
+
+        if try_rpc_node_first {
```

Comment:
It'll be nice to refactor some of the logic in a separate function, it seems like this function is getting bigger and harder to follow. Also, could we update the docstring of this function?

Reply:
I made a pass, and hopefully it's cleaner now. PTAL.